### PR TITLE
fix(web): removes package namespacing from kbd's CSS class

### LINKED ISSUE #3910: bug(web/engine): incorrect CSS classes for keyboards with embedded namespacing
> I inspected on my Android device and saw the following class for the 'khmer_angkor' keyboard: `kmw-keyboard-khmer_angkor::khmer_angkor`. I don't know why we're getting that class name and it smells like a bug @jahorton?
> 
> That was with 14.0.184-alpha.

No, that sounds right - that's how keyboards are 'namespaced' in the embedded context, at least on Android.  To be clear, that's `kmw-keyboard-<package id>::<keyboard id>`, resulting from a keyboard's full ID being `<package id>::<keyboard id>`, a decision @darcywong00 made when we first started utilizing the KMP format as part of the Android implementation.  I'd have to double-check whether or not that's mirrored within the iOS engine.

That said, I guess it's a bug that we didn't think to filter it and strip it down to just the `<keyboard_id>` portion afterward for the CSS classes.

_Originally posted by @jahorton in https://github.com/keymanapp/keyboards/issues/1388#issuecomment-730848173_

As a minor recap of notes seen there:

https://github.com/keymanapp/keyman/blob/f16872a99896a2fc208c68ac92a827820ab07e3f/web/source/kmwembedded.ts#L51-L56

overrides the original keyboard ID, which results in the CSS class form seen above.  That CSS was originally written under the assumption that the keyboard ID would be appended to the string `kmw-keyboard-`, which keyboard developers may then rely upon.  The issue is that the namespacing breaks the association under embedded Android; we'd need to check for the namespacing and revert it within that context.  Should be a relatively simple fix.

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #3910.

### CODE REVIEW

File: web/source/osk/oskManager.ts [hunk 1]
```diff
@@ -261,8 +261,14 @@ namespace com.keyman.osk {
       }
 
       // Correct the classname for the (inner) OSK frame (Build 360)
+      var kbdID: string = (activeKeyboard ? activeKeyboard.id.replace('Keyboard_','') : '');
+      if(keymanweb.isEmbedded && kbdID.indexOf('::') != -1) {
+        // De-namespaces the ID for use with CSS classes.
+        // Assumes that keyboard IDs may not contain the ':' symbol.
```
Comment (mcdurdin) [inline web/source/osk/oskManager.ts:267]: Keyboard IDs have to be valid Javascript identifiers. So punctuation is not possible. Also, we restrict the names to `[A-Za-z0-9_]+` anyway (and disallow digits as first letter)

**Review by mcdurdin — APPROVED** [review]

File: web/source/osk/oskManager.ts (hunk 1 above)
Comment (jahorton) [reply web/source/osk/oskManager.ts:267]: That's what I thought.  Always best to make the assumptions explicit just in case, though.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.247-beta](https://keyman.com/downloads/releases/beta/14.0.247)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.19-alpha](https://keyman.com/downloads/releases/alpha/15.0.19)

**Comment by MattGyverLee** [comment]
I think this is in the current release versions. Is it safe to remove the extra namespacing from my CSS?

**Comment by MattGyverLee** [comment]
@mcdurdin @darcywong00  I think this is in the current release versions. Is it safe to remove the extra namespacing from my CSS?

**Comment by mcdurdin** [comment]
> Is it safe to remove the extra namespacing from my CSS?

This would only be safe if you target version 14.0 in your keyboard, which would mean it would not be available to users on older versions. So probably not ideal.

**Pull request merged**